v0/destinations/yahoo_dsp/util.js

```diff
@@ -0,0 +1,173 @@
+const qs = require("qs");
+const { generateJWTToken } = require("../../../util/jwtTokenGenerator");
+const { httpSend, httpPOST } = require("../../../adapters/network");
+const { CustomError } = require("../../util");
+
+const { ACCESS_TOKEN_CACHE_TTL } = require("./config.js");
+const Cache = require("../../util/cache");
+
+const accessTokenCache = new Cache(ACCESS_TOKEN_CACHE_TTL);
+
+const getUnixTimestamp = () => {
+  return Math.floor(Date.now() / 1000);
+};
+
+/**
+ * There are four location types which are stored in the poiLocationType error. These are ["chains", "woeids", "gids", "categories"].
+ * The ones to be added in includes Object will be taken from audienceType with include as Prefix. The ones to be added in excludes 
+ * Object will be taken from audienceType with exclude as Prefix.   
+ */
+const poiLocationType = [
+  "includeChains",
+  "includeWoeids",
+  "includeCategories",
+  "includeGids",
+  "excludeChains",
+  "excludeWoeids",
+  "excludeCategories",
+  "excludeGids"
+];
+let listType;
+const includes = {};
+const excludes = {};
+
+// updating includes 
+const populateIncludes = (list, audienceType) => {
+  list.forEach(element => {
+    const keys = Object.keys(element);
+    keys.forEach(elementKey => {
+      if (poiLocationType.includes(elementKey)) {
+        listType = audienceType;
+      }
+    });
+    if (!listType) {
+      throw new CustomError(`Audience type ${audienceType} not provided`, 400);
+    }
+    if (element.includeChains) {
+      if (!includes.chains) {
+        includes.chains = [];
+      }
+      includes.chains.push(element.includeChains);
+    }
+    if (element.includeWoeids) {
+      if (!includes.woeids) {
+        includes.woeids = [];
+      }
+      includes.woeids.push(element.includeWoeids);
+    }
+    if (element.includeGids) {
+      if (!includes.gids) {
+        includes.gids = [];
+      }
+      includes.gids.push(element.includeGids);
+    }
+    if (element.includeCategories) {
+      if (!includes.categories) {
+        includes.categories = [];
+      }
+      includes.categories.push(element.includeCategories);
+    }
+  });
+  return includes;
```

Comment:
Use switch-case here.

Reply:
Here, we are checking for the properties if it is defined then pushing it in an array. It will be very complex with switch case statements.

Reply:
Got it.